### PR TITLE
Bug 1209650 - Add plumbing to camera app to support pause/resume duri…

### PR DESCRIPTION
**Pull request by aosmond**
…ng video recording.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ac4eea746daf4c20f07ffec595a2b781d8bc00da

File: apps/camera/js/lib/camera/camera.js [hunk 1]
```diff
@@ -713,6 +713,21 @@ Camera.prototype.capture = function(options) {
 };
 
 /**
+ * Pauses/resumes a video capture session.
+ *
+ * @public
+ */
+Camera.prototype.pauseCapture = function() {
+  if (!this.mozCamera || this.mode !== 'video') {
+    return false;
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: Does this need to return a value? You're not returning anything below.

File: apps/camera/js/lib/camera/camera.js [hunk 2]
```diff
@@ -951,6 +966,43 @@ Camera.prototype.startedRecording = function() {
...
+  if (state !== 'started' && state !== 'resumed') {
+    debug('not recording, cannot pause');
+    return;
+  }
+
+  this.mozCamera.pauseRecording();
+  this.set('recording', 'pausing');
+};
+
+Camera.prototype.pausedRecording = function() {
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: I know there's no convention currently for this, but if would be nice if things like this that are essentially callbacks were named clearer as such. Like `onRecordingPaused`.

File: apps/camera/js/lib/camera/camera.js [hunk 3]
```diff
@@ -1251,8 +1307,13 @@ Camera.prototype.isMode = function(mode) {
  *
  * @private
  */
-Camera.prototype.startVideoTimer = function() {
-  this.set('videoStart', new Date().getTime());
+Camera.prototype.startVideoTimer = function(resume) {
+ if (resume) {
+    var delta = this.videoStopped - this.get('videoStart');
+    this.set('videoStart', (new Date().getTime()) - delta);
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: Replace `new Date.getTime()` with `Date.now()`. It avoids creating a new instance of a `Date` object when all you're after is the epoch number.

Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: Also, when you change this to `Date.now()`, the extra set of parenthesis are not needed :-)

File: apps/camera/js/lib/camera/camera.js [hunk 4]
```diff
@@ -1251,8 +1307,13 @@ Camera.prototype.isMode = function(mode) {
...
+  } else {
+    this.set('videoStart', new Date().getTime());
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: `Date.now()`

File: apps/camera/js/lib/camera/camera.js [hunk 5]
```diff
@@ -1265,6 +1326,7 @@ Camera.prototype.startVideoTimer = function() {
 Camera.prototype.stopVideoTimer = function() {
   clearInterval(this.videoTimer);
   this.videoTimer = null;
+  this.videoStopped = new Date().getTime();
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: `Date.now()`

File: apps/camera/js/lib/camera/camera.js [hunk 6]
```diff
@@ -1103,6 +1155,10 @@ Camera.prototype.onRecorderStateChange = function(e) {
     this.emit('filesizelimitreached');
   } else if(msg === 'Started') {
     this.startedRecording();
+  } else if(msg === 'Paused') {
```
Comment (justindarc) [inline apps/camera/js/lib/camera/camera.js]: nit: should be a space after `if` everywhere here

**Pull request closed without merging**